RDKShell/RDKShell.cpp

```diff
@@ -236,6 +238,16 @@ namespace WPEFramework {
                    serviceCallsign.append(".1");
                    gSystemServiceConnection = getThunderControllerClient(serviceCallsign);
                 }
+                else if (currentState == PluginHost::IShell::ACTIVATED && service->Callsign() == RESIDENTAPP_CALLSIGN && sResidentAppFirstLaunch)
+                {
+                    sResidentAppFirstLaunch = false;
+                    bool launchFactoryApp = mShell.checkForBootupFactoryAppLaunch();
+                    if (launchFactoryApp)
+                    {
+                      JsonObject request, response;
+                      uint32_t status = getThunderControllerClient("org.rdk.RDKShell.1")->Invoke(0, "launchFactoryApp", request, response);
```

Comment:
We should rework this to avoid a thunder api call.  for example something like: `mShell.launchFactoryApp()`